src/binding/c/attr_api.txt

```diff
@@ -0,0 +1,202 @@
+    for which a pointer does not fit in an integer (e.g., any > 32 bit address
+    system that uses 64 bits for Fortran 'DOUBLE PRECISION').
+
+    If an attribute is already present, the delete function (specified when the
+    corresponding keyval was created) will be called.
+
+*/
+
+MPI_Comm_create_keyval:
+    .desc: Create a new attribute key
+    .seealso: MPI_Comm_free_keyval
+/*
+    Notes:
+    Key values are global (available for any and all communicators).
+
+    Default copy and delete functions are available.  These are
+    + MPI_COMM_NULL_COPY_FN   - empty copy function
+    . MPI_COMM_NULL_DELETE_FN - empty delete function
+    - MPI_COMM_DUP_FN         - simple dup function
+
+    There are subtle differences between C and Fortran that require that the
+    copy_fn be written in the same language from which 'MPI_Comm_create_keyval'
+    is called.
+    This should not be a problem for most users; only programmers using both
+    Fortran and C in the same program need to be sure that they follow this rule.
+
+    .N AttrErrReturn
+*/
+
+MPI_Comm_delete_attr:
+    .desc: Deletes an attribute value associated with a key on
+    .seealso: MPI_Comm_set_attr, MPI_Comm_create_keyval
+    .extra: ignore_revoke
+
+MPI_Comm_free_keyval:
+    .desc: Frees an attribute key for communicators
+{
+    MPIR_free_keyval(comm_keyval_ptr);
+    *comm_keyval = MPI_KEYVAL_INVALID;
+}
+
+MPI_Comm_get_attr:
+    .desc: Retrieves attribute value by key
+    .polymorph: MPIR_Attr_type attr_type=MPIR_ATTR_PTR
+    .extra: ignore_revoke
+
+MPI_Comm_set_attr:
+    .desc: Stores attribute value associated with a key
+    .seealso: MPI_Comm_create_keyval, MPI_Comm_delete_attr
+    .polymorph: MPIR_Attr_type attr_type=MPIR_ATTR_PTR
+    .extra: ignore_revoke
+/*
+    Notes:
+    Values of the permanent attributes 'MPI_TAG_UB', 'MPI_HOST', 'MPI_IO',
+    'MPI_WTIME_IS_GLOBAL', 'MPI_UNIVERSE_SIZE', 'MPI_LASTUSEDCODE', and
+    'MPI_APPNUM' may not be changed.
+
+    The type of the attribute value depends on whether C, C++, or Fortran
+    is being used.
+    In C and C++, an attribute value is a pointer ('void *'); in Fortran, it is an
+    address-sized integer.
+
+    If an attribute is already present, the delete function (specified when the
+    corresponding keyval was created) will be called.
+*/
+
+MPI_Keyval_create:
+    .desc: Greates a new attribute key
+    .replace: deprecated with MPI_Comm_create_keyval
+    .seealso: MPI_Keyval_free, MPI_Comm_create_keyval
+/*
+    Notes:
+    Key values are global (available for any and all communicators).
+
+    There are subtle differences between C and Fortran that require that the
+    copy_fn be written in the same language that 'MPI_Keyval_create'
+    is called from.
+    This should not be a problem for most users; only programmers using both
+    Fortran and C in the same program need to be sure that they follow this rule.
```

Comment:
`MPI_Comm_create_keyval()` has `.N AttrErrReturn` in `Notes` while `MPI_Keyval_create()` does not. Do we need to include it in the comment?

Reply:
`MPI_Keyval_create` is deprecated. I would hate to maintain two copies of man page notes (currently we do). I think I would prefer to make all deprecated function man page very abbreviated (i.e. strip the extra notes) and just refer to it is deprecated and point user to the replacement function, where they can read all the notes. What do you think?

Reply:
For the sake of reviewing, I'm fine if there's a rule to check these changes.  For example, I'm fine if removing all notes and comments is the goal of this PR.

I prefer a more complete description for each function (maybe copied from the spec) like Open MPI, but sure, we can omit the details for some deprecated functions. To be honest, I don't know those notes in MPICH describe either implementation-defined behavior (which is MPICH-specific) or some miscellaneous things that are specified by the specification.

https://www.mpich.org/static/docs/v3.3/www3/MPI_Keyval_create.html
https://www.open-mpi.org/doc/v4.0/man3/MPI_Keyval_create.3.php

Reply:
My current goal is to just reproduce our current man pages plus some easy consistency or simplifications. I agree with you that we should have more details in the man pages, but a full scale man page sweep will be a separate project.

Reply:
fixed in #4955